src/software/ai/passing/cost_function.cpp

```diff
@@ -11,159 +11,128 @@
+double rateShot(const Point& shot_origin, const Field& field, const Team& enemy_team,
+                const TbotsProto::PassingConfig& passing_config)
+{
+    auto shot_opt =
+        calcBestShotOnGoal(Segment(field.enemyGoalpostPos(), field.enemyGoalpostNeg()),
+                           shot_origin, enemy_team.getAllRobots(), TeamType::ENEMY);
 
     Angle open_angle_to_goal = Angle::zero();
-    Point shot_target        = field.enemyGoalCenter();
     if (shot_opt && shot_opt.value().getOpenAngle().abs() > Angle::fromDegrees(0))
     {
         open_angle_to_goal = shot_opt.value().getOpenAngle();
     }
 
-    // Figure out what the maximum open angle of the goal could be from the receiver pos.
-    Angle goal_angle = convexAngle(field.enemyGoalpostNeg(), pass.receiverPoint(),
-                                   field.enemyGoalpostPos())
-                           .abs();
-    double net_percent_open = 0;
-    if (goal_angle > Angle::zero())
-    {
-        net_percent_open = open_angle_to_goal.toDegrees() / goal_angle.toDegrees();
-    }
+    const double min_ideal_angle =
+        passing_config.min_ideal_pass_shoot_goal_open_angle_deg();
+    double open_angle_to_goal_score = open_angle_to_goal.toDegrees();
 
-    // Create the shoot score by creating a sigmoid that goes to a large value as
-    // the section of net we're shooting on approaches 100% (ie. completely open)
-    double shot_openness_score = sigmoid(net_percent_open, 0.45, 0.95);
-
-    // Prefer angles where the robot does not have to turn much after receiving the
-    // pass to take the shot (or equivalently the shot deflection angle)
-    //
-    // Receiver robots on the friendly side, almost always, need to rotate a full 180
-    // degrees to shoot on net. So we relax that requirement for both receiver and ball
-    // locations on the friendly side
-    //
-    // TODO (#1987) This creates a very steep slope, find a better way to do this
-    if (pass.receiverPoint().x() < 0 || pass.passerPoint().x() < 0)
-    {
-        ideal_max_rotation_to_shoot_degrees = 180;
-    }
-    Angle rotation_to_shot_target_after_pass = pass.receiverOrientation().minDiff(
-        (shot_target - pass.receiverPoint()).orientation());
-    double required_rotation_for_shot_score =
-        1 - sigmoid(rotation_to_shot_target_after_pass.abs().toDegrees(),
-                    ideal_max_rotation_to_shoot_degrees, 4);
+    // Clamp angle to [0, min_ideal_angle], where all angle >=min_ideal_angle are given
+    // a score of 1.0.
+    open_angle_to_goal_score = std::clamp(open_angle_to_goal_score, 0.0, min_ideal_angle);
+
+    // Linearly scale score to [0.0, 1.0]
+    open_angle_to_goal_score = open_angle_to_goal_score / min_ideal_angle;
+
+    // Linearly scale score to [min_pass_shoot_score, 1.0] to stop this cost function
+    // from returning a very low score, causing the other cost functions to be ignored.
+    return open_angle_to_goal_score;
```

Comment:
Similarly here, shots are rated using the open angle as opposed to the openness percentage. With the updated implementation, 0 degrees receives a rating of 0, and it linearly increases to a rating of 1 when the net is open >10 degrees.